hummingbird/ml/containers/sklearn/tvm_containers.py

```diff
@@ -176,17 +182,31 @@ def load_param_dict(param_bytes):
 
         return container
 
-    def _to_tvm_tensor(self, *inputs):
-        tvm_tensors = {}
-        msg = "The number of input rows {} is different from the batch size {} the TVM model is compiled for."
-        for i, inp in enumerate(inputs):
-            assert inp.shape[0] == self._batch_size, msg.format(inp.shape[0], self._batch_size)
-            tvm_tensors[self._input_names[i]] = self._to_tvm_array(inp)
-        return tvm_tensors
-
     def _predict_common(self, output_index, *inputs):
-        self.model.run(**self._to_tvm_tensor(*inputs))
-        return self.model.get_output(output_index).asnumpy()
+        # Compute padding.
+        padding_size = 0
+        if self._pad_input:
+            padding_size = self._batch_size - inputs[0].shape[0]
+
+        # Prepare inputs.
+        for i, input_ in enumerate(inputs):
+            assert (
+                self._pad_input or input_.shape[0] == self._batch_size
+            ), "The number of input rows {} is different from the batch size {} the TVM model is compiled for.".format(
```

Comment:
In what case would this fail?  If you're doing `padding_size = self._batch_size - inputs[0].shape[0]` above, then isn't this guaranteed to be true.   (Assuming not garbage input I guess?)

Reply:
It should fail when pad is false and `input_.shape[0]` and `self._batch_size` are different, no?

Reply:
Ah!  now i see that TVM_PAD_PREDICTION_INPUT can be false, got it